security/pkg/nodeagent/plugin/providers/google/stsclient/stsclient.go

```diff
@@ -106,7 +106,7 @@ func (p *SecureTokenServiceExchanger) requestWithRetry(reqBytes []byte) ([]byte,
 			break
 		}
 		monitoring.NumOutgoingRetries.With(monitoring.RequestType.Value(monitoring.TokenExchange)).Increment()
-		stsClientLog.Debugf("token exchange request failed: status code %v, body %v", resp.StatusCode, string(body))
+		stsClientLog.Errorf("token exchange request failed: status code %v, body %v", resp.StatusCode, string(body))
 		time.Sleep(p.backoff)
 	}
 	return nil, fmt.Errorf("exchange failed all retries, last error: %v", lastError)
```

Comment:
can we fix this to properly return the error? Currently this only works if httpRequest.Do returns an error which is not that common, more likely it returns a non-200 error code in which case lastError is nil

Reply:
Hm I see you mentioned offline it will handle non-200 code. But I am, not sure why this logs `last error: <nil>` in some cases. Shouldn't that be impossible?

Reply:
last error: <nil> appears only in the 1.9 branch. In master-asm branch, it will be impossible.